bidict/_loose.py

```diff
@@ -0,0 +1,9 @@
+from ._bidict import bidict
+
+class loosebidict(bidict):
+    """
+    A mutable bidict which always uses forcing put operations
+    so that it never raises :class:`ValueExistsException`.
+    """
+    def _put(self, key, val):
```

Comment:
Originally had that but changed it in favor of dynamic superclass lookup
for more extensibility. Worth it?

On Wednesday, December 2, 2015, Daniel Pope <notifications@github.com>
wrote:

> In bidict/_loose.py
> <https://github.com/jab/bidict/pull/26#discussion_r46466346>:
>
> > @@ -0,0 +1,9 @@
> > +from ._bidict import bidict
> > +
> > +class loosebidict(bidict):
> > +    """
> > +    A mutable bidict which always uses forcing put operations
> > +    so that it never raises :class:`ValueExistsException`.
> > +    """
> > +    def _put(self, key, val):
>
> Tiny performance optimisation, but
>
> _put = bidict.forceput
>
> would remove function call overhead without increasing code duplication. I
> imagine _put() will get called a lot in loops, so this adds up.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jab/bidict/pull/26/files#r46466346>.
>